sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -797,10 +865,30 @@ private void downloadToFileCleanup(AsynchronousFileChannel channel) {
      *
      * @return A reactive response signalling completion.
      */
-    public Mono<VoidResponse> undelete() {
+    public Mono<Void> undelete() {
+        return undeleteWithResponse().flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Undelete restores the content and metadata of a soft-deleted blob and/or any associated soft-deleted snapshots.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobAsyncClient.undeleteWithResponse}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/undelete-blob">Azure Docs</a></p>
+     *
+     * @return A reactive response signalling completion.
+     */
+    public Mono<VoidResponse> undeleteWithResponse() {
+        return withContext(context -> undeleteWithResponse(context));
+    }
```

Comment:
I wonder if there is much value in the 'withResponse' variant when the return type of the non-withResponse variant is Mono<Void>. I wonder if in the case of Mono<Void> return types, we just change it to Mono<VoidResponse> and get rid of the 'withResponse' variety. I would worry a bit about discoverability and inconsistency, but on the upside, this lowers the method count. Thoughts?

Reply:
I think in addition to consistency, those `withResponse` return type methods could be useful to the users to know the status of the operation.

Reply:
Sorry, GitHub stripped out the generic typing in my previous comment. I've fixed it - please re-read it to see what I actually meant to say :-) The Response information is still present in my suggestion - but I do totally accept it is not consistent, hence why I'm seeking your opinion :-)

Reply:
yeah, I think I agree with this. I don't think `Mono<Void>` apart from consistency add any real benefits to the user.
Should I make this change as well?